src/models/utils/index.ts

```diff
@@ -19,3 +22,129 @@ export function onlyHasFields(obj: object, fields: string[]): boolean {
 export function isFlagEnabled(Flags: number, checkFlag: number): boolean {
   return (checkFlag & Flags) === checkFlag;
 }
+
+/**
+ * Sets a transaction's flags to its numeric representation
+ *
+ * @param {Transaction} tx A transaction to set its flags to its numeric representation
+ * @returns {void}
+ */
+export function setTransactionFlagsToNumber(tx: Transaction): void {
+  switch (tx.TransactionType) {
+    case 'OfferCreate':
+      tx.Flags = convertOfferCreateFlagsToNumber(tx.Flags as OfferCreateFlags);
+      break;
+    case 'PaymentChannelClaim':
+      tx.Flags = convertPaymentChannelClaimFlagsToNumber(tx.Flags as PaymentChannelClaimFlags);
+      break;
+    case 'Payment':
+      tx.Flags = convertPaymentTransactionFlagsToNumber(tx.Flags as PaymentTransactionFlags);
+      break;
+    case 'TrustSet':
+      tx.Flags = convertTrustSetFlagsToNumber(tx.Flags as TrustSetFlags);
+      break;
+    default:
+      tx.Flags = convertGlobalFlagsToNumber(tx.Flags);
+  }
+}
+
+function convertGlobalFlagsToNumber(flags: GlobalFlags | number | undefined): number {
+  if (typeof flags === 'number') {
+    return flags;
+  }
+  return flags?.tfFullyCanonicalSig ? GlobalFlagsEnum.tfFullyCanonicalSig : 0;
+}
+
+function convertOfferCreateFlagsToNumber(flags: OfferCreateFlags): number {
+  if (flags == null) {
+    return 0;
+  } else if (typeof flags === 'number') {
+    return flags;
+  }
+
+  let resultFlags = convertGlobalFlagsToNumber(flags);
+
+  if (flags.tfPassive) {
+    resultFlags |= OfferCreateFlagsEnum.tfPassive;
+  }
+  if (flags.tfImmediateOrCancel) {
+    resultFlags |= OfferCreateFlagsEnum.tfImmediateOrCancel;
+  }
+  if (flags.tfFillOrKill) {
+    resultFlags |= OfferCreateFlagsEnum.tfFillOrKill;
+  }
+  if (flags.tfSell) {
+    resultFlags |= OfferCreateFlagsEnum.tfSell;
+  }
+
+  return resultFlags;
+}
+
+function convertPaymentChannelClaimFlagsToNumber(flags: PaymentChannelClaimFlags): number {
+  if (flags == null) {
+    return 0;
+  } else if (typeof flags === 'number') {
+    return flags;
+  }
+
+  let resultFlags = convertGlobalFlagsToNumber(flags);
+
+  if (flags.tfRenew) {
+    resultFlags |= PaymentChannelClaimFlagsEnum.tfRenew;
+  }
+  if (flags.tfClose) {
+    resultFlags |= PaymentChannelClaimFlagsEnum.tfClose;
+  }
+
+  return resultFlags;
+}
+
+function convertPaymentTransactionFlagsToNumber(flags: PaymentTransactionFlags): number {
+  if (flags == null) {
+    return 0;
+  } else if (typeof flags === 'number') {
+    return flags;
+  }
```

Comment:
this is now repeated a few times. i think you should factor this out into its own function